tasks/main.yml

```diff
@@ -95,62 +137,56 @@
       register: mounted_ISO
       when: mount_path is defined and not mount_path.stdout
 
-    - name: Check if VBoxGuest additions ISO is mounted
-      shell: mount -l 2>/dev/null|awk 'tolower($0) ~ /vbox.*additions/{print $3}'
-      args:
-        warn: False
-      register: mount_path
-
-    - name: Find out architecture
-      shell: uname -r | cut -d'-' -f3
-      register: kernel
-      changed_when: not kernel.stdout
+    - name: Save the current list of packages
+      shell: "{{ package_list_command }} > /tmp/before.txt"
+      when: virtualbox_keep is undefined or not virtualbox_keep
+      changed_when: false
 
-    - name: Save the current list of packages for Debians
-      block:
-        - name: Query current list
-          shell: dpkg-query -l > /tmp/before.txt
-          when: virtualbox_keep is undefined or not virtualbox_keep
-
-        - name: Install kernel headers for Debian
-          apt:
-            name: linux-headers-{{ kernel.stdout }}
-            update_cache: yes
-            cache_valid_time: 86400
-            install-recommends: no
-            state: present
-          when: kernel is defined and kernel.stdout is defined
+    - name: Get running kernel version
+      shell: "uname -r"
```

Comment:
When no output redirection is necessary, it's safer to use the ``command`` module instead of the ``shell`` module.